pkg/parcacol/ingest.go

```diff
@@ -254,22 +254,18 @@ func NormalizeWriteRawRequest(ctx context.Context, normalizer Normalizer, req *p
 
 		samples := make([][]*profile.NormalizedProfile, 0, len(rawSeries.Samples))
 		for _, sample := range rawSeries.Samples {
-			r, err := gzip.NewReader(bytes.NewBuffer(sample.RawProfile))
-			if err != nil {
-				return NormalizedWriteRawRequest{}, status.Errorf(codes.Internal, "failed to create gzip reader: %v", err)
-			}
-
-			content, err := io.ReadAll(r)
-			if err != nil {
-				return NormalizedWriteRawRequest{}, status.Errorf(codes.InvalidArgument, "failed to decompress profile: %v", err)
-			}
-
-			if err := r.Close(); err != nil {
-				return NormalizedWriteRawRequest{}, status.Errorf(codes.Internal, "failed to close gzip reader: %v", err)
+			if len(sample.RawProfile) >= 2 && sample.RawProfile[0] == 0x1f && sample.RawProfile[1] == 0x8b {
+				gz, err := gzip.NewReader(bytes.NewBuffer(sample.RawProfile))
```

Comment:
Is this missing the `defer gz.Close()` now?

Reply:
you're right!